media-libs/svt-av1/svt-av1-9999.ebuild

```diff
@@ -21,26 +21,51 @@ fi
 LICENSE="BSD-2 Apache-2.0 BSD ISC LGPL-2.1+ MIT"
 SLOT="0/$(ver_cut 1)"
 
-BDEPEND="amd64? ( dev-lang/yasm )"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="amd64? ( dev-libs/cpuinfo )"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	amd64? ( dev-lang/yasm )
+	test? ( dev-cpp/gtest-parallel )
+"
 
 PATCHES=(
 	"${FILESDIR}"/${PN}-1.5.0-fortify-no-override.patch
 )
 
-multilib_src_configure() {
-	append-ldflags -Wl,-z,noexecstack
+src_prepare() {
+	cmake_src_prepare
+
+	# Lets not install tests
+	sed -e '/install(/d' -i test/CMakeLists.txt || die
+
+	# Needs more setup to run in the ebuild
+	cmake_run_in test cmake_comment_add_subdirectory api_test
+	# Tries to downloads stuff for this test
+	cmake_run_in test cmake_comment_add_subdirectory e2e_test
+}
 
+multilib_src_configure() {
 	local mycmakeargs=(
-		# Tests require linking against https://github.com/Cidana-Developers/aom/tree/av1-normative ?
-		# undefined reference to `ifd_inspect'
-		# https://github.com/Cidana-Developers/aom/commit/cfc5c9e95bcb48a5a41ca7908b44df34ea1313c0
-		# .. and https://gitlab.com/AOMediaCodec/SVT-AV1/-/blob/master/.gitlab/workflows/linux/.gitlab-ci.yml implies it's all quite manual?
-		-DBUILD_TESTING=OFF
+		# Upstream only supports 64-bit and specially amd64 and arm64.
+		# Other enviroments will fail to build due to missing symbols.
+		-DBUILD_TESTING=$(multilib_native_usex test)
 		-DCMAKE_OUTPUT_DIRECTORY="${BUILD_DIR}"
-		-DENABLE_AVX512=ON
+		-DUSE_CPUINFO=SYSTEM # will only be used on amd64
 	)
 
 	[[ ${ABI} != amd64 ]] && mycmakeargs+=( -DCOMPILE_C_ONLY=ON )
 
 	cmake_src_configure
 }
+
+multilib_src_test() {
+	if multilib_is_native_abi; then
+		# Upstream uses this, and this gives a significant time save in running these tests.
+		# 2025-05-19T19:39:25 >>> media-libs/svt-av1-3.0.2: 1:46:14
+		# 2025-05-20T16:10:34 >>> media-libs/svt-av1-3.0.2: 20′35″
+		edo gtest-parallel --workers $(makeopts_jobs) "${BUILD_DIR}"/SvtAv1UnitTests
```

Comment:
Haven't tried it yet but, out of curiosity given it likely does heavy AV1 encoding, is the ram usage for this kind of what you'd expect? (aka less than 2GB per job)

Reply:
I haven't noticed anything unreasonable. Looks like around 20-30MB per job now that I look at it.

Reply:
No issue then, been a while since I tried svt-av1 (some years ago before ffmpeg had support for it) and wasn't sure.